### PR TITLE
Fix an issue with buffering codec that was sending a poison event down

### LINKED ISSUE #19: Codec multiliner send rogue events down to the pipeline
From: https://discuss.elastic.co/t/multiline-codec-is-not-working-with-filebeats-plugin-in-logstash-2-0-0/33768/3

Aggregating events with the multiline codec can cause the plugin to send rogue events down the pipeline.

```
Default settings used: Filter workers: 2
Beats: SSL Certificate will not be used {:level=>:warn}
Beats: SSL Key will not be used {:level=>:warn}
Logstash startup completed
NoMethodError: undefined method `to_hash_with_metadata' for ["0"]:Array
  encode_with_metadata at /Users/ph/es/logstash/vendor/bundle/jruby/1.9/gems/logstash-codec-rubydebug-2.0.3/lib/logstash/codecs/rubydebug.rb:38
                  call at org/jruby/RubyMethod.java:120
                encode at /Users/ph/es/logstash/vendor/bundle/jruby/1.9/gems/logstash-codec-rubydebug-2.0.3/lib/logstash/codecs/rubydebug.rb:30
               receive at /Users/ph/es/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-stdout-2.0.2/lib/logstash/outputs/stdout.rb:54
                handle at /Users/ph/es/logstash/lib/logstash/outputs/base.rb:78
           output_func at (eval):22
          outputworker at /Users/ph/es/logstash/lib/logstash/pipeline.rb:252
         start_outputs at /Users/ph/es/logstash/lib/logstash/pipeline.rb:169
```

logstash config
```
input {
  beats {
    port => 5555
    codec => multiline {
      max_lines => 2000
      pattern => "\d+"
      negate => true
      what => "previous"
    }
    ssl_certificate => "/Users/ph/es/logstash-config/certificates/logstash-forwarder.crt"
    ssl_key => "/Users/ph/es/logstash-config/certificates/logstash-forwarder.key"
  }
}
output {
  stdout { codec  => rubydebug { metadata => true } }
}
```
Sample log
```
70572
something else
70573
something else
70574
something else
7057
```

### PR DESCRIPTION
**Pull request by ph**
Fix an issue with buffering codec that was sending a poison event down

Fixes #19

### CODE REVIEW

**Comment by ph** [comment]
@jsvd Can you review this?

File: spec/inputs/beats_spec.rb [hunk 1]
```diff
@@ -120,6 +120,17 @@
           expect(event["tags"]).to include("bonjour")
         end
       end
+
+      context "when data is buffered in the codec" do
+        let(:codec) { LogStash::Codecs::Multiline.new("pattern" => '^\s', "what" => "previous") }
+        let(:event_map) { {"message" => "hello?", "tags" => ["syslog"]} }
+
+        it "retuns nil" do
+          puts beats.codec
```
Comment (jsvd) [inline spec/inputs/beats_spec.rb]: :129 dd

**Comment by ph** [comment]
updated with your comment, added changelog + gemspec bump

**Comment by jsvd** [comment]
LGTM

**Pull request closed without merging**